src/service.rs

```diff
@@ -69,21 +64,23 @@ pub struct GlobalState<Types: NodeType> {
     >,
 
     // registered builer states
-    pub spawned_builder_states: HashMap<VidCommitment, Types::Time>,
+    pub spawned_builder_states:
+        HashMap<(VidCommitment, Types::Time), BroadcastSender<MessageType<Types>>>,
 
     // builder state -> last built block , it is used to respond the client
     // if the req channel times out during get_available_blocks
-    pub builder_state_to_last_built_block: HashMap<VidCommitment, ResponseMessage>,
+    pub builder_state_to_last_built_block: HashMap<(VidCommitment, Types::Time), ResponseMessage>,
 
     // scheduled GC by view number
-    pub view_to_cleanup_targets:
-        BTreeMap<Types::Time, (Vec<VidCommitment>, Vec<BuilderCommitment>)>,
-
-    // sending a request from the hotshot to the builder states
-    pub request_sender: BroadcastSender<MessageType<Types>>,
+    pub view_to_cleanup_targets: BTreeMap<
+        Types::Time,
+        (
+            Vec<VidCommitment>,
+            Vec<(VidCommitment, BuilderCommitment, Types::Time)>,
```

Comment:
This is begging for some structs...we have a mapping whoes value is a tuple of vecs of tuples, `VidCommitment` is appearing twice in the value and `Types::Time` appears in both the key and the value. Not clear whether this is redundant or the same type is being used to refer to different things

Reply:
https://github.com/EspressoSystems/hotshot-builder-core/pull/120/commits/08c8e9d95a57d738a240a91c86511a04dd384aa7

https://github.com/EspressoSystems/hotshot-builder-core/pull/120/commits/d3530562868f4297fd9b118eb8b95d382512f542